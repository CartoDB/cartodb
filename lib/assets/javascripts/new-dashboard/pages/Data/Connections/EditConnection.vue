<template>
  <Dialog ref="dialog"
    :headerTitle="connectionsSuccessfulId ?
      $t('ConnectorsPage.successConnection') :
      (editing ? $t('ConnectorsPage.editConnection', {connector: title}) : $t('ConnectorsPage.addConnection'))"
    :headerImage="connectionsSuccessfulId ?
      require('../../../assets/icons/datasets/conected.svg') :
      (editing ? require('../../../assets/icons/datasets/edit-connection.svg') : require('../../../assets/icons/datasets/add-connection.svg'))"
    :showSubHeader="!editing && !connectionsSuccessfulId"
    :backRoute="{name: backNamedRoute}"
  >
    <template slot="sub-header">
      <h3 class="title is-caption is-regular is-txtMidGrey u-flex u-flex__align--center" v-if="!editing">
        <img height="21" class="u-mr--8" :src="logo">
        {{ $t('DataPage.connectWith') }} {{title}}
      </h3>
    </template>
    <template #default>

      <template v-if="!connectionsSuccessfulId">
        <DatabaseConnectionForm v-if="type === 'database' && !isBigQuery"
          :connector="importOption"
          :connection="connection"
          @connectClicked="connectionSuccess" @cancel="onCancel"></DatabaseConnectionForm>
        <BigQuerySelectionMode v-else-if="isBigQuery"
          :connection="connection" @cancel="onCancel" @connectionSuccess="connectionSuccess"></BigQuerySelectionMode>
        <OAuthConnectionForm v-else-if="type === 'cloud'"
          :connector="importOption"
          :connection="connection" @connectionSuccess="connectionSuccess"></OAuthConnectionForm>
      </template>

      <div v-else-if="connectionsSuccessfulId" class="connections-successfull u-flex u-flex__direction--column u-flex__align--center">
        <h3 class="is-semibold u-mt--16">{{ $t('DataPage.connectionsSuccessfull') }}</h3>
        <h4 class="is-caption u-mt--16 is-regular">{{$t('DataPage.imports.database.successTitle', { brand: title })}}</h4>
        <div class="u-mt--48 u-flex u-flex__align--center">
          <button @click="onCancel" class="button is-primary">{{ $t('Dialogs.finish') }}</button>
          <!-- <router-link :to="{ name: 'your-connections' }">
            <button class="is-small is-semibold is-txtPrimary">{{ $t('DataPage.imports.connections') }}</button>
          </router-link>
          <router-link v-if="isBigQuery" :to="{ name: 'tilesets' }">
            <button class="u-ml--36 is-small is-semibold is-txtPrimary">{{$t('DataPage.tilesets')}}</button>
          </router-link>
          <button @click="navigateNext" class="button is-primary u-ml--36">{{ $t('DataPage.addYourData') }}</button> -->
        </div>
      </div>
    </template>
  </Dialog>
</template>

<script>

import exportedScssVars from 'new-dashboard/styles/helpers/_assetsDir.scss';
import Dialog from 'new-dashboard/components/Dialogs/Dialog.vue';
import { getImportOption } from 'new-dashboard/utils/connector/import-option';
import BigQuerySelectionMode from 'new-dashboard/components/Connector/BigQuerySelectionMode';
import OAuthConnectionForm from 'new-dashboard/components/Connector/OAuthConnectionForm';
import DatabaseConnectionForm from 'new-dashboard/components/Connector/DatabaseConnectionForm';
import { mapState } from 'vuex';

export default {
  name: 'EditConnection',
  components: {
    Dialog,
    DatabaseConnectionForm,
    OAuthConnectionForm,
    BigQuerySelectionMode
  },
  props: {
    backNamedRoute: {
      default: ''
    }
  },
  data () {
    return {
      connectionsSuccessfulId: null
    };
  },
  computed: {
    ...mapState({
      rawConnections: state => state.connectors.connections
    }),
    isBigQuery () {
      return this.importOption && this.importOption.options && this.importOption.options.service === 'bigquery';
    },
    importOption () {
      let connector = this.editing ? (this.connection ? this.connection.connector : null) : this.$route.params.connector;
      const option = getImportOption(connector);
      return option;
    },
    editing () {
      return this.$route.name === 'edit-connection';
    },
    connection () {
      return this.rawConnections && this.editing ? this.rawConnections.find(conn => conn.id === this.$route.params.id) : null;
    },
    logo () {
      return this.importOption && `${exportedScssVars.assetsDir && exportedScssVars.assetsDir.replace(/\"/g, '')}/images/layout/connectors/${this.importOption.name}.svg`;
    },
    title () {
      return this.importOption && this.importOption.title;
    },
    type () {
      return this.importOption && this.importOption.type;
    }
  },
  methods: {
    connectionSuccess (id) {
      this.connectionsSuccessfulId = id;
      this.$store.dispatch('connectors/fetchConnectionsList');
    },
    onCancel () {
      this.$refs.dialog.closePopup();
    },
    navigateNext () {
      const routeNamePrefix = !this.editing ? this.$route.name.replace('connector-selected', '') : 'new-connection-';
      this.$router.push({ name: `${routeNamePrefix}connection-dataset`, params: { id: this.connectionsSuccessfulId } });
    }
  }
};
</script>

<style scoped lang="scss">
@import "new-dashboard/styles/variables";
h3.title {
  white-space: nowrap;
}
.connections-successfull {
  h3 {
    font-size: 24px;
    line-height: 1.5;
  }
}
</style>
