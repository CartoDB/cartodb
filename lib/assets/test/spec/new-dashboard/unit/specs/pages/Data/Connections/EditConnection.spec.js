import Vuex from 'vuex';
import { createLocalVue, mount } from '@vue/test-utils';
import EditConnection from 'new-dashboard/pages/Data/Connections/EditConnection';

const localVue = createLocalVue();
localVue.use(Vuex);

const MYSQL = {
  id: 'ID',
  connector: 'mysql',
  name: 'MySQL',
  type: 'db-connector',
  connectionParams: {
    server: 'server',
    port: 5432,
    database: 'database',
    username: 'username',
    password: 'password'
  }
};

describe('Connections.vue', () => {
  beforeEach(() => {
  });

  describe('Connection type "DATABASE"', () => {
    it('should render edit version', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: [ MYSQL ]
        }
      });
      const connectionsList = createConnections(store, 'edit-connection');

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render create version', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: [ MYSQL ]
        }
      });
      const connectionsList = createConnections(store);

      expect(connectionsList).toMatchSnapshot();
    });
  });

  describe('Connection type "CLOUD"', () => {
    it('should render create version properly', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: []
        }
      });
      const connectionsList = createConnections(store, '', {
        connector: 'gdrive'
      });

      expect(connectionsList).toMatchSnapshot();
    });
  });

  describe('Connection type "BQ"', () => {
    it('should render create version properly', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: []
        }
      });
      const connectionsList = createConnections(store, '', {
        connector: 'bigquery'
      });

      expect(connectionsList).toMatchSnapshot();
    });
  });
});

function createStore (customStoreData) {
  const storeData = {
    connectors: {
      connections: null,
      projects: null,
      bqDatasets: null,
      loadingConnections: false,
      loadingProjects: true
    },
    config: {
      isFirstTimeViewingDashboard: false
    },
    user: {
      account_type: 'account_type'
    }
  };

  const storeInstance = new Vuex.Store({
    state: customStoreData || storeData
  });

  return storeInstance;
}

function createConnections (store, routeName = '', routeParams = {
  id: 'ID',
  connector: 'mysql'
}) {
  return mount(EditConnection, {
    mocks: {
      $t: key => key,
      backboneViews: {},
      $route: {
        name: routeName,
        params: routeParams
      },
      $router: {
        push: jest.fn()
      }
    },
    store,
    localVue
  });
}
