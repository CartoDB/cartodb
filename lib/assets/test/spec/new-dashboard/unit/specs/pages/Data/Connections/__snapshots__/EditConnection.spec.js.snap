// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connections.vue Connection type "BQ" should render create version properly 1`] = `
<div class="Dialog is-white"><button class="CDB-Shape Dialog-closeBtn"><img src="../../assets/icons/common/close.svg"></button>
  <div class="Dialog-contentWrapper Dialog-contentWrapper--withHeaderWrapper">
    <div class="Dialog-header CreateDialog-header with-separator"><img height="36" src="[object Object]" class="u-mb--12">
      <h2 class="CDB-Text CDB-Size-large u-mainTextColor">ConnectorsPage.addConnection</h2>
    </div>
    <div class="Dialog-body Dialog-body--expanded Dialog-body--create Dialog-body--noPaddingTop Dialog-body--withoutBorder Dialog-body--no-spacing">
      <div class="container grid">
        <div class="grid-cell grid-cell--col12 u-flex u-flex__direction--column">
          <div class="u-flex u-flex__justify--center u-flex__align--center u-pt--10 sub-header"><button class="back is-small is-semibold is-txtPrimary u-flex u-flex__align--center"><img src="../../assets/icons/common/icon-prev-blue.svg" class="u-mr--8">
              Connect
            </button>
            <h3 class="title is-caption is-regular is-txtMidGrey u-flex u-flex__align--center"><img height="21" src="undefined/images/layout/connectors/bigquery.svg" class="u-mr--8">
              DataPage.connectWith BigQuery
            </h3>
          </div>
          <div class="u-mt--32">
            <div class="u-flex u-flex__justify--center">
              <div class="u-flex u-flex__justify--center">
                <div class="main u-flex u-flex__align--center u-flex__direction--column">
                  <div class="disclaimer">
                    <h4 class="is-small is-semibold u-mt--16">ConnectorsPage.BigQuery.title</h4>
                    <ul class="u-mt--8">
                      <li class="u-mb--16"><span class="text is-small">ConnectorsPage.BigQuery.disclaimer1</span></li>
                      <li class="u-mb--16"><span class="text is-small">ConnectorsPage.BigQuery.disclaimer2</span></li>
                      <li class="u-mb--16"><span class="text is-small">ConnectorsPage.BigQuery.disclaimer3</span></li>
                      <li class="u-mb--16"><span class="text is-small">ConnectorsPage.BigQuery.disclaimer4</span></li>
                    </ul>
                    <div class="u-mt--48 u-flex u-flex__justify--center u-flex__align--center"><button class="is-small is-semibold is-txtPrimary">ConnectorsPage.BigQuery.conditions.cancel</button> <button class="button is-primary u-ml--36">DataPage.continue</button></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="footer"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Connections.vue Connection type "CLOUD" should render create version properly 1`] = `
<div class="Dialog is-white"><button class="CDB-Shape Dialog-closeBtn"><img src="../../assets/icons/common/close.svg"></button>
  <div class="Dialog-contentWrapper Dialog-contentWrapper--withHeaderWrapper">
    <div class="Dialog-header CreateDialog-header with-separator"><img height="36" src="[object Object]" class="u-mb--12">
      <h2 class="CDB-Text CDB-Size-large u-mainTextColor">ConnectorsPage.addConnection</h2>
    </div>
    <div class="Dialog-body Dialog-body--expanded Dialog-body--create Dialog-body--noPaddingTop Dialog-body--withoutBorder Dialog-body--no-spacing">
      <div class="container grid">
        <div class="grid-cell grid-cell--col12 u-flex u-flex__direction--column">
          <div class="u-flex u-flex__justify--center u-flex__align--center u-pt--10 sub-header"><button class="back is-small is-semibold is-txtPrimary u-flex u-flex__align--center"><img src="../../assets/icons/common/icon-prev-blue.svg" class="u-mr--8">
              Connect
            </button>
            <h3 class="title is-caption is-regular is-txtMidGrey u-flex u-flex__align--center"><img height="21" src="undefined/images/layout/connectors/gdrive.svg" class="u-mr--8">
              DataPage.connectWith Google Drive
            </h3>
          </div>
          <div class="u-mt--32">
            <div class="u-flex u-flex__direction--column u-flex__align--center">
              <!----> <span class="text is-small is-txtMidGrey">ConnectorsPage.oauthRequesting</span>
              <!---->
              <div class="u-mt--32">
                <div class="loading-state">
                  <div class="loading-state-icon"><img svg-inline="" src="../../assets/icons/common/loading.svg" class="loading-state__svg primary" style="height: 40px; width: 40px;"></div>
                  <h6 class="loading-state-text text is-caption is-txtSoftGrey"></h6>
                </div>
              </div>
            </div>
          </div>
          <div class="footer"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Connections.vue Connection type "DATABASE" should render create version 1`] = `
<div class="Dialog is-white"><button class="CDB-Shape Dialog-closeBtn"><img src="../../assets/icons/common/close.svg"></button>
  <div class="Dialog-contentWrapper Dialog-contentWrapper--withHeaderWrapper">
    <div class="Dialog-header CreateDialog-header with-separator"><img height="36" src="[object Object]" class="u-mb--12">
      <h2 class="CDB-Text CDB-Size-large u-mainTextColor">ConnectorsPage.addConnection</h2>
    </div>
    <div class="Dialog-body Dialog-body--expanded Dialog-body--create Dialog-body--noPaddingTop Dialog-body--withoutBorder Dialog-body--no-spacing">
      <div class="container grid">
        <div class="grid-cell grid-cell--col12 u-flex u-flex__direction--column">
          <div class="u-flex u-flex__justify--center u-flex__align--center u-pt--10 sub-header"><button class="back is-small is-semibold is-txtPrimary u-flex u-flex__align--center"><img src="../../assets/icons/common/icon-prev-blue.svg" class="u-mr--8">
              Connect
            </button>
            <h3 class="title is-caption is-regular is-txtMidGrey u-flex u-flex__align--center"><img height="21" src="undefined/images/layout/connectors/mysql.svg" class="u-mr--8">
              DataPage.connectWith MySQL
            </h3>
          </div>
          <div class="u-mt--32">
            <div class="u-flex u-flex__justify--center">
              <div>
                <!---->
                <h4 class="is-small is-semibold">DataPage.imports.database.title</h4>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper"><label class="is-small u-mr--16">DataPage.imports.database.label-name</label> <input type="text" placeholder="DataPage.imports.database.placeholder-name"></div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">server</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-server" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">port</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-port" type="number">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">database</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-database" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">username</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-username" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">password</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-password" type="password">
                </div>
                <!---->
                <div class="u-flex u-flex__justify--end u-mt--32">
                  <!----> <button class="CDB-Button CDB-Button--primary CDB-Button--big is-disabled"><span class="u-flex CDB-Text is-semibold CDB-Size-medium"><!---->
          DataPage.connect
        </span></button></div>
              </div>
              <div class="info u-ml--80 ">
                <h4 class="is-small is-semibold u-mb--8">DataPage.gettingConnected</h4>
                <p class="text u-mt--10 is-txtMidGrey is-small">
                  DataPage.connectInfo
                </p>
                <div class="ports u-pt--16 u-pb--16 u-pl--24 u-pr--24 u-mt--16 is-txtMidGrey is-small u-flex u-flex__direction--column u-flex__justify--between"><span class="u-mb--4">35.188.111.175</span> <span class="u-mb--4">35.192.121.157</span> <span class="u-mb--4">35.238.166.16</span> <span class="u-mb--4">35.224.241.96</span> <span class="u-mb--4">35.232.84.56</span> <span class="u-mb--4">35.205.127.169</span> <span class="u-mb--4">35.233.31.201</span> <span class="u-mb--4">35.187.80.96</span> <span class="u-mb--4">104.197.28.44</span> <span class="u-mb--4">104.199.3.149</span> <span>104.155.126.89</span></div>
              </div>
            </div>
          </div>
          <div class="footer"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Connections.vue Connection type "DATABASE" should render edit version 1`] = `
<div class="Dialog is-white"><button class="CDB-Shape Dialog-closeBtn"><img src="../../assets/icons/common/close.svg"></button>
  <div class="Dialog-contentWrapper Dialog-contentWrapper--withHeaderWrapper">
    <div class="Dialog-header CreateDialog-header with-separator"><img height="36" src="[object Object]" class="u-mb--12">
      <h2 class="CDB-Text CDB-Size-large u-mainTextColor">ConnectorsPage.editConnection</h2>
    </div>
    <div class="Dialog-body Dialog-body--expanded Dialog-body--create Dialog-body--noPaddingTop Dialog-body--withoutBorder Dialog-body--no-spacing">
      <div class="container grid">
        <div class="grid-cell grid-cell--col12 u-flex u-flex__direction--column">
          <!---->
          <div class="u-mt--32">
            <div class="u-flex u-flex__justify--center">
              <div>
                <div class="alert u-pl--36 u-pt--8 u-pb--8 u-pr--8 u-mb--24"><span class="is-small">ConnectorsPage.editConnectionDisclaimer</span></div>
                <h4 class="is-small is-semibold">DataPage.imports.database.title</h4>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper"><label class="is-small u-mr--16">DataPage.imports.database.label-name</label> <input type="text" placeholder="DataPage.imports.database.placeholder-name"></div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">server</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-server" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">port</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-port" type="number">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">database</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-database" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">username</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-username" type="text">
                </div>
                <div class="u-flex u-flex__align--center u-flex__justify--between u-mt--24 input-wrapper">
                  <div class="u-flex u-flex__direction--column u-flex__align--end u-flex__grow--1  u-mr--16"><label class="is-small">password</label>
                    <!---->
                  </div> <input placeholder="DataPage.imports.database.placeholder-password" type="password">
                </div>
                <!---->
                <div class="u-flex u-flex__justify--end u-mt--32"><button class="u-mr--28 is-small is-semibold is-txtPrimary">ConnectorsPage.cancel</button> <button class="CDB-Button CDB-Button--primary CDB-Button--big is-disabled"><span class="u-flex CDB-Text is-semibold CDB-Size-medium"><!---->
          ConnectorsPage.editConnectionButton
        </span></button></div>
              </div>
              <div class="info u-ml--80 ">
                <h4 class="is-small is-semibold u-mb--8">DataPage.gettingConnected</h4>
                <p class="text u-mt--10 is-txtMidGrey is-small">
                  DataPage.connectInfo
                </p>
                <div class="ports u-pt--16 u-pb--16 u-pl--24 u-pr--24 u-mt--16 is-txtMidGrey is-small u-flex u-flex__direction--column u-flex__justify--between"><span class="u-mb--4">35.188.111.175</span> <span class="u-mb--4">35.192.121.157</span> <span class="u-mb--4">35.238.166.16</span> <span class="u-mb--4">35.224.241.96</span> <span class="u-mb--4">35.232.84.56</span> <span class="u-mb--4">35.205.127.169</span> <span class="u-mb--4">35.233.31.201</span> <span class="u-mb--4">35.187.80.96</span> <span class="u-mb--4">104.197.28.44</span> <span class="u-mb--4">104.199.3.149</span> <span>104.155.126.89</span></div>
              </div>
            </div>
          </div>
          <div class="footer"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
